pkg/scheduler/objects/queue.go

```diff
@@ -1258,6 +1258,48 @@ func (sq *Queue) GetMaxResource() *resources.Resource {
 	return sq.internalGetMax(limit)
 }
 
+// GetFairMaxResource computes the fair max resources for a given queue.
+// Starting with the root, descend down to the target queue allowing children to override Resource values .
+// If the root includes an explicit 0 value for a Resource, do not include it in the accumulator and treat it as missing.
+// If no children provide a maximum capacity override, the resulting value will be the value found on the Root.
+// It is useful for fair-scheduling to allow a ratio to be produced representing the rough utilization % of a given queue.
+func (sq *Queue) GetFairMaxResource() *resources.Resource {
+	var limit *resources.Resource
+	if sq.parent == nil {
+		cleaned := resources.NewResource()
+		if sq.maxResource == nil {
+			return cleaned
+		}
+
+		for k, v := range sq.maxResource.Resources {
+			if v != 0 {
+				cleaned.Resources[k] = v
+			}
+		}
+
+		return cleaned
+	}
+
+	limit = sq.parent.GetFairMaxResource()
+	return sq.internalGetFairMaxResource(limit)
+}
+
+func (sq *Queue) internalGetFairMaxResource(limit *resources.Resource) *resources.Resource {
+	sq.RLock()
+	defer sq.RUnlock()
+
+	if sq.maxResource.IsEmpty() || limit.IsEmpty() {
+		return limit
```

Comment:
This should be `return limit.Clone()`. `Resource` objects are not immutable, so it's better to clone since we're returning one.

Reply:
done

Reply:
Also, to address Wilfred's comment earlier about using limit directly rather than cloning -- yes, that's safe *today* but the fact we're passing in limit to this function means a future caller might be rather surprised to find that limit is both mutated and returned from the function. I'd rather sacrifice a tiny bit of performance than risk unexpected updates.